ksp_plugin_test/plugin_compatibility_test.cpp

```diff
@@ -113,17 +126,109 @@ class PluginCompatibilityTest : public testing::Test {
                                       preferred_compressor,
                                       preferred_encoder);
   }
+
+  static void CheckSaveCompatibility(std::filesystem::path const& filename,
+                              std::string_view const compressor,
+                              std::string_view const encoder) {
+    // Read a plugin from the given file.
+    auto plugin = ReadPluginFromFile(filename, compressor, encoder);
+
+    WriteAndReadBack(std::move(plugin));
+  }
 };
 
 TEST_F(PluginCompatibilityTest, PreCartan) {
   // This space for rent.
 }
 
 TEST_F(PluginCompatibilityTest, PreCohen) {
+  CaptureStderr();
   CheckSaveCompatibility(
-      SOLUTION_DIR / "ksp_plugin_test" / "pre_cohen.proto.hex",
+      SOLUTION_DIR / "ksp_plugin_test" / "saves" / "3039.proto.hex",
       /*compressor=*/"",
       /*decoder=*/"hexadecimal");
+  EXPECT_THAT(
+      GetCapturedStderr(),
+      AllOf(
+          HasSubstr(
+              "pre-Cohen ContinuousTrajectory"),  // Regression test for #3039.
+          HasSubstr("pre-Cauchy"),                // The save is even older.
+          Not(HasSubstr("pre-Cartan"))            // But not *that* old.
+          ));
+}
+
+TEST_F(PluginCompatibilityTest, Reach) {
+  CaptureStderr();
+  not_null<std::unique_ptr<Plugin const>> plugin = ReadPluginFromFile(
+      SOLUTION_DIR / "ksp_plugin_test" / "saves" / "3072.proto.b64",
+      /*compressor=*/"gipfeli",
+      /*decoder=*/"base64");
+  EXPECT_THAT(GetCapturedStderr(),
+              AllOf(HasSubstr("pre-Galileo"), Not(HasSubstr("pre-Frobenius"))));
+  auto const test = plugin->GetVessel("f2d77873-4776-4809-9dfb-de9e7a0620a6");
+  EXPECT_THAT(test->name(), Eq("TEST"));
+  EXPECT_THAT(TTSecond(test->psychohistory().front().time),
+              Eq("1970-08-14T08:03:18"_DateTime));
+  EXPECT_THAT(TTSecond(test->psychohistory().back().time),
+              Eq("1970-08-14T08:47:05"_DateTime));
+  EXPECT_THAT(test->has_flight_plan(), IsFalse());
+
+  auto const infnity =
+      plugin->GetVessel("29142a79-7acd-47a9-a34d-f9f2a8e1b4ed");
+  EXPECT_THAT(infnity->name(), Eq("IFNITY-5.2"));
+  EXPECT_THAT(TTSecond(infnity->psychohistory().front().time),
+              Eq("1970-08-14T08:03:46"_DateTime));
+  EXPECT_THAT(TTSecond(infnity->psychohistory().back().time),
+              Eq("1970-08-14T08:47:05"_DateTime));
+  EXPECT_THAT(infnity->has_flight_plan(), IsTrue());
```

Comment:
`EXPECT_TRUE`, or rather `ASSERT_TRUE` (the next statement would die if this is false.)